### PR TITLE
Fixed #422 Issues with dateFormatter docs

### LINKED ISSUE #422: Issues with dateFormatter docs
There's several issues with https://github.com/jquery/globalize/blob/0eebc186c426c59f7b0376dfbdcbf1ec4caacbf1/doc/api/date/date-formatter.md (noticed while reviewing #417)

* [x] The default argument is named "pattern", the Parameters section also says "pattern", but then describes it as a skeleton. Should rename the argument to "skeleton" to reflect its value, if that is indeed correct
* [x] The description for skeleton, a little bit further down, says "String value indicating a skeleton (see description above)", but there's actually very little description above. The key is the "date field" link (link text should be improved) that points at the UTS#35 docs for the skeleton fields. We should have a bit more information of our own, since the external doc is rather overwhelming.
* [x] `pattern` is warned against - maybe we should reflect that in the name, e.g. rename to `rawPattern`? Or `machinePattern` to indicate that its not for human consumption? If yes, should create a separate issue for that
* [x] Missing linebreak after "date" option - the others all have their description on a separate line
* [x] "Their possible values are: full, long, medium, or short." - replace the "or" with "and"
* [x] For the last section, "For improved performance on iterations"
  * Does that really belong here? Seems like something that the `formatDate()` method should mention/warn about.
  * If we keep this, make it executable: Replace or define `a` and `b` variables, replace or remove the `...`, do something with the formatted array.

### PR DESCRIPTION
**Pull request by ManrajGrover**
@rxaviers sir, @jzaefferer sir,

I have fixed Points 1,3,4,5,6 of issues mentioned in #422 . Please review and let me know the changes and final decision of Point 2 mentioned in the same issue.

### CODE REVIEW

File: doc/api/date/date-formatter.md [hunk 1]
```diff
@@ -141,10 +142,10 @@ For improved performance on iterations, first create the formatter. Then, reuse
 it on each loop.
 
 ```javascript
-var dates = [ new Date( a ), new Date( b ), ... ];
+var dates = [ new Date( 2010, 10, 30, 17, 55 ), new Date( 2015, 3, 18, 4, 25 )]; // Array can be very large
```
Comment (rxaviers) [inline doc/api/date/date-formatter.md]: Space
```diff
- )];
+ ) ];
```

Comment (rxaviers) [inline doc/api/date/date-formatter.md]: @jzaefferer do you suggest any comment to express the "..."? Should we drop it?

File: doc/api/date/date-formatter.md [hunk 2]
```diff
@@ -141,10 +142,10 @@ For improved performance on iterations, first create the formatter. Then, reuse
...
 var formatter = Globalize( "en" ).dateFormatter({ time: "short" });
-
 formattedDates = dates.map(function( date ) {
   return formatter( date );
 });
+// > Array [ "5:55 PM", "4:25 AM" ]
```
Comment (rxaviers) [inline doc/api/date/date-formatter.md]: :+1:

File: doc/api/date/date-formatter.md [hunk 3]
```diff
@@ -141,10 +142,10 @@ For improved performance on iterations, first create the formatter. Then, reuse
 it on each loop.
 
 ```javascript
-var dates = [ new Date( a ), new Date( b ), ... ];
+var dates = [ new Date( 2010, 10, 30, 17, 55 ), new Date( 2015, 3, 18, 4, 25 )]; // Array can be very large
 var formatter = Globalize( "en" ).dateFormatter({ time: "short" });
-
```
Comment (rxaviers) [inline doc/api/date/date-formatter.md]: Any reason for dropping this empty line?

File: doc/api/date/date-formatter.md (hunk 2 above)
Comment (rxaviers) [inline doc/api/date/date-formatter.md]: Not your fault. But, we need to update the below as well.

```diff
- formattedDates = dates.map(function( date ) {
+ var formattedDates = dates.map(function( date ) {
```

File: doc/api/date/date-formatter.md (hunk 1 above)
Comment (ManrajGrover) [inline doc/api/date/date-formatter.md]: @rxaviers sir,

Fixed. Should I change the comment?

File: doc/api/date/date-formatter.md (hunk 2 above)
Comment (ManrajGrover) [inline doc/api/date/date-formatter.md]: Fixed sir.

File: doc/api/date/date-formatter.md (hunk 3 above)
Comment (ManrajGrover) [inline doc/api/date/date-formatter.md]: @rxaviers sir,

Was just trying to save space. Should I revert the change?

Comment (rxaviers) [inline doc/api/date/date-formatter.md]: Yeap, please and thanks in advance.

Comment (ManrajGrover) [inline doc/api/date/date-formatter.md]: @rxaviers sir,

Done.

File: doc/api/date/date-formatter.md (hunk 1 above)
Comment (rxaviers) [inline doc/api/date/date-formatter.md]: I'm waiting for an input from @jzaefferer...

Comment (rxaviers) [inline doc/api/date/date-formatter.md]: I'm moving this question up...

**Comment by rxaviers** [comment]
@jzaefferer do you suggest any comment to express the "..."? Should we drop it?

File: doc/api/date/date-formatter.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ be formatted.
 
 ### Parameters
 
-**pattern**
+**machinePattern**
```
Comment (rxaviers) [inline doc/api/date/date-formatter.md]: Please, can you revert this change? The documentation needs to be updated only when we in fact perform this change.

File: doc/api/date/date-formatter.md (hunk 1 above)
Comment (ManrajGrover) [inline doc/api/date/date-formatter.md]: Sure sir.

File: doc/api/date/date-formatter.md (hunk 4 above)
Comment (ManrajGrover) [inline doc/api/date/date-formatter.md]: @rxaviers sir, 

Changed.

**Comment by rxaviers** [comment]
Awesome thanks

**Comment by ManrajGrover** [comment]
@jzaefferer sir @rxaviers sir,

Lets finalize and land this fix. Thank you.

File: doc/api/date/date-formatter.md [hunk 5]
```diff
@@ -141,10 +142,11 @@ For improved performance on iterations, first create the formatter. Then, reuse
 it on each loop.
 
 ```javascript
-var dates = [ new Date( a ), new Date( b ), ... ];
+var dates = [ new Date( 2010, 10, 30, 17, 55 ), new Date( 2015, 3, 18, 4, 25 ) ]; // Array can be very large
```
Comment (jzaefferer) [inline doc/api/date/date-formatter.md]: Line comment in front

**Comment by jzaefferer** [comment]
Just the one comment issue, rest looks good. Need one more follow-up ticket from #422 before that can be considered fixed.

File: doc/api/date/date-formatter.md (hunk 5 above)
Comment (jzaefferer) [inline doc/api/date/date-formatter.md]: To be more specific, and using a more helpful comment:

```js
// In an application, this array could have a few hundred entries
var dates = [ ...
```

**Comment by rxaviers** [comment]
> Just the one comment issue, rest looks good. Need one more follow-up ticket from #422 before that can be considered fixed.

Let's have this PR landed saying "Ref #422". The two follow-up tickets from #422 can be fixed in upcoming PRs.

**Comment by ManrajGrover** [comment]
@rxaviers sir, @jzaefferer sir,

Made the comment change.

**Comment by ManrajGrover** [comment]
@jzaefferer sir, @rxaviers sir,

Please review it and let me know the changes. Thank you. :)

**Comment by rxaviers** [comment]
LGTM. Thanks
